apps/whale-api/src/module.api/consortium.service.ts

```diff
@@ -1,17 +1,135 @@
+      tokenAmounts: tx.amounts.map((a: any) => {
+        const splits = a.split('@')
+        return { token: splits[1], amount: splits[0] }
+      }),
+      txId: tx.txid,
+      address: tx.owner,
+      block: tx.blockHeight
+    }
+  }
+
+  private isValidTxIdFormat (value: string): boolean {
+    return /^[0-9a-f]{64}$/.exec(value) !== null
+  }
+
+  private async getFilteredUniqueMembers (searchTerm: string): Promise<MemberDetail[]> {
+    const attrs = (await this.rpcClient.masternode.getGov('ATTRIBUTES')).ATTRIBUTES
+    const membersKeyRegex: RegExp = /^v0\/consortium\/\d+\/members$/
+    const searchForMemberDetail = searchTerm !== '' && !this.isValidTxIdFormat(searchTerm)
+
+    return (Object.entries(attrs) as [[string, object]]).reduce((prev: MemberDetail[], [key, value]) => {
+      if (membersKeyRegex.exec(key) === null) {
+        return prev
+      }
+
+      (Object.entries(value) as [[string, MemberDetail]]).forEach(([memberId, memberDetail]) => {
+        if (searchForMemberDetail) {
+          if (!(memberDetail.ownerAddress === searchTerm || memberDetail.name.toLowerCase().includes(searchTerm))) {
+            return prev
+          }
+        }
+
+        if (!prev.some(m => m.id === memberId)) {
+          prev.push({
+            id: memberId,
+            name: memberDetail.name,
+            ownerAddress: memberDetail.ownerAddress
+          })
+        }
+      })
+
+      return prev
+    }, [])
+  }
+
+  private async getSingleHistoryTransactionResponse (searchTerm: string, members: MemberDetail[]): Promise<ConsortiumTransactionResponse> {
+    const foundTx = await this.transactionMapper.get(searchTerm)
+    if (foundTx === undefined) {
+      return {
+        transactions: [],
+        total: 0
+      }
+    }
+
+    const relevantTxsOnBlock = await this.rpcClient.account.listAccountHistory(members.map(m => m.ownerAddress), {
+      maxBlockHeight: foundTx.block.height,
+      depth: 0,
+      txtypes: [DfTxType.MINT_TOKEN, DfTxType.BURN_TOKEN]
+    })
+    const transaction = relevantTxsOnBlock.find(tx => tx.txid === foundTx.txid)
+    if (transaction === undefined) {
+      return {
+        transactions: [],
+        total: 0
+      }
+    }
+
+    return {
+      transactions: [this.formatTransactionResponse(transaction, members)],
+      total: 1
+    }
+  }
+
+  private async getPaginatedHistoryTransactionsResponse (pageIndex: number, limit: number, members: MemberDetail[]): Promise<ConsortiumTransactionResponse> {
+    const memberAddresses = members.map(m => m.ownerAddress)
+
+    const transactions: AccountHistory[] = await this.rpcClient.account.listAccountHistory(memberAddresses, {
+      txtypes: [DfTxType.MINT_TOKEN, DfTxType.BURN_TOKEN],
+      including_start: true,
+      start: pageIndex * limit,
+      limit
+    })
+
+    const totalTxCount = await this.rpcClient.account.historyCount(memberAddresses, { txtypes: [DfTxType.BURN_TOKEN, DfTxType.MINT_TOKEN] })
```

Comment:
I assume this is a `LevelDB:Scan` all records operation, and very expensive to call. Can we avoid this or find alternatives?

Reply:
For each API call, there will be maximum 2 rpc calls eventhough we have 3 `this.rpcClient.account.*` usages, as the `getPaginatedHistoryTransactionsResponse` terminates the flow.

About alternatives: I couldn't find find an alternative to get total count and transactions in terms of RPCs thus I used the combination of these `historyCount` and `listAccountHistory`.  Though, Maybe I should take a look if I can do something with the indexer?!